direct/io-elastic/src/main/java/cz/o2/proxima/direct/elastic/ElasticWriter.java

```diff
@@ -0,0 +1,160 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.elastic;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonObject;
+import cz.o2.proxima.direct.core.CommitCallback;
+import cz.o2.proxima.direct.core.OnlineAttributeWriter;
+import cz.o2.proxima.repository.AttributeDescriptor;
+import cz.o2.proxima.storage.StreamElement;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.bulk.BulkProcessor;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.common.xcontent.XContentType;
+
+@Slf4j
+public class ElasticWriter implements OnlineAttributeWriter, BulkProcessor.Listener {
+  private final ElasticAccessor accessor;
+  private final RestHighLevelClient client;
+  private final Map<IndexRequest, CommitCallback> callbacksToCommit = new ConcurrentHashMap<>();
+  private final BulkProcessor bulkProcessor;
+
+  public ElasticWriter(ElasticAccessor accessor) {
+    this.accessor = accessor;
+    this.client = accessor.getRestHighLevelClient();
+    this.bulkProcessor =
+        BulkProcessor.builder(
+                (request, bulkListener) ->
+                    client.bulkAsync(request, RequestOptions.DEFAULT, bulkListener),
+                this)
+            .setBulkActions(accessor.getBatchSize())
+            .setConcurrentRequests(accessor.getConcurrentRequests())
+            .setBulkSize(new ByteSizeValue(10, ByteSizeUnit.MB))
+            .build();
+  }
+
+  @Override
+  public URI getUri() {
+    return accessor.getUri();
+  }
+
+  @Override
+  public void write(StreamElement element, CommitCallback commitCallback) {
+    Preconditions.checkArgument(!element.isDelete(), "Delete not supported.");
+    Preconditions.checkArgument(
+        !element.getAttributeDescriptor().isWildcard(), "Wildcard not supported.");
```

Comment:
What is the reason for this? I don't see anything that would stop it from working.

Reply:
I think ... it wasn't needed and so I didn't test it. I don't think there is any other reason.